cmake/Modules/dotnet/UseCSharpProjectBuilder.cmake

```diff
@@ -154,26 +154,27 @@ function(csharp_add_project name)
         -DMSBUILD_TOOLSET="${MSBUILD_TOOLSET}"
         -DCSHARP_IMPORTS="${CSHARP_IMPORTS}"
         -DCONFIG_INPUT_FILE="${CSBUILD_CSPROJ_IN}"
-        -DCONFIG_OUTPUT_FILE="${CURRENT_TARGET_BINARY_DIR}/${CSBUILD_${name}_CSPROJ}"
+        -DCONFIG_OUTPUT_FILE="${CURRENT_TARGET_BINARY_DIR}/${name}/${CSBUILD_${name}_CSPROJ}"
         -P ${dotnet_cmake_module_DIR}/ConfigureFile.cmake
 
         COMMAND ${CMAKE_COMMAND}
         -DCSHARP_PACKAGE_REFERENCES="${CSHARP_LEGACY_PACKAGE_REFERENCES}"
         -DCONFIG_INPUT_FILE="${dotnet_cmake_module_DIR}/Modules/dotnet/packages.config.in"
-        -DCONFIG_OUTPUT_FILE="${CURRENT_TARGET_BINARY_DIR}/packages.config"
+        -DCONFIG_OUTPUT_FILE="${CURRENT_TARGET_BINARY_DIR}/${name}/packages.config"
         -P ${dotnet_cmake_module_DIR}/ConfigureFile.cmake
 
         COMMAND ${CMAKE_COMMAND}
-        -DCONFIG_INPUT_FILE="${dotnet_cmake_module_DIR}/Modules/dotnet/Directory.Build.props"
-        -DCONFIG_OUTPUT_FILE="${CURRENT_TARGET_BINARY_DIR}/Directory.Build.props"
+        -DCSHARP_BUILDER_OUTPUT_NAME="${name}${CSBUILD_OUTPUT_SUFFIX}"
+        -DCONFIG_INPUT_FILE="${dotnet_cmake_module_DIR}/Modules/dotnet/Directory.Build.props.in"
+        -DCONFIG_OUTPUT_FILE="${CURRENT_TARGET_BINARY_DIR}/${name}/${name}$/Directory.Build.props"
```

Comment:
This feels like a typo based on the pattern in the rest of the changes, but I haven't thought/tested in great detail.

Did you mean to have ${name}/${name}$/Directory.Build.props?

Reply:
Been playing around with the change -- I do think this is a typo. Resulting build directory:

![image](https://user-images.githubusercontent.com/1154591/80304306-ec68a280-8782-11ea-9d98-ade7c46572e7.png)

Reply:
Exactly. This is a typo

Reply:
Pardon me being pedantic, but just confirming the duplicate name (`${name}/${name}`) is intentional? Or should the Directory.Build.props file be in the root along with the csproj?

Again I don't know this stuff too well; I only ask because everywhere else you've only added one level of subdirectory.

Reply:
I have changed it, and everything is working, and tests are passing. I thought that it was because of some concurrency problems that I had while building. I have removed it, and I hope not to revert this change in the future...

Reply:
If you think there was a reason for it, I believe you and go ahead and leave it.

Again, my comments are mostly based on pattern matching in the diffs, not deep analysis of the build tree!